autoscale_cloudroast/test_repo/autoscale/fixtures.py

```diff
@@ -455,61 +452,69 @@ def get_server_list():
                 group_id).entity
             if group_state.desiredCapacity == desired_capacity:
                 server_list = get_server_list()
                 if (len(server_list) == expected_servers):
                     return server_list
-            time.sleep(5)
-        else:
-            server_list = get_server_list()
+
             self.fail(
-                'Waited 2 mins for desired capacity/active server list to '
-                'reach the server count of {0}. Has desired capacity {1} on '
-                'the group {2} and {3} servers on the account. '
+                'Waited {0} secs for desired capacity/active server list to '
+                'reach the server count of {1}. Has desired capacity {2} on '
+                'the group {3} and {4} servers on the account. '
                 'Filtering by server_name={server_name}'.format(
+                    elapsed_time,
                     desired_capacity,
                     group_state.desiredCapacity, group_id,
                     len(server_list),
                     server_name=server_name))
 
-    def assert_servers_deleted_successfully(self, server_name, count=0):
+        return self.autoscale_behaviors.retry(
+            check_servers, timeout=120, interval_time=5, time_scale=time_scale)
+
+    def assert_servers_deleted_successfully(self, server_name, count=0,
+                                            time_scale=True):
         """
-        Given a partial server name, polls for 15 mins to assert that the tenant id
-        has only specified count of servers containing that name, and returns the list
-        of servers.
+        Given a partial server name, polls for 15 mins to assert that the
+        tenant id has only specified count of servers containing that name,
+        and returns the list of servers.
         """
-        endtime = time.time() + 900
-        while time.time() < endtime:
+        def check_deleted(elapsed_time):
             server_list = self.get_servers_containing_given_name_on_tenant(
                 server_name=server_name)
             if len(server_list) == count:
                 return server_list
-            time.sleep(self.interval_time)
-        else:
-            self.fail('Servers on the tenant with name {0} were not deleted even'
-                      ' after waiting 15 mins'.format(server_name))
+            self.fail('Servers on the tenant with name {0} were not deleted '
+                      'even after waiting {1} seconds'.format(
+                          server_name, elapsed_time))
+        return self.autoscale_behaviors.retry(
+            check_deleted,  timeout=900, time_scale=time_scale)
 
     def delete_nodes_in_loadbalancer(self, node_id_list, load_balancer):
         """
         Given the node id list and load balancer id, check for lb status
         'PENDING UPDATE' and delete node when lb is ACTIVE
         """
+        class _TimeoutError(Exception):
```

Comment:
Weird, I don't remember why I did this instead of `self.fail` like I did everywhere else - would it be ok if I just changed it to the `self.fail` instead? Retrying will just raise the last error if it can't retry anymore, right?